ytypes/common.go

```diff
@@ -385,21 +385,24 @@ func getJSONTreeValForField(parentSchema, schema *yang.Entry, f reflect.StructFi
 // getJSONTreeValForPath returns a JSON subtree from tree at the given path from
 // the root. If returns (nil, false) if no subtree is found at the given path.
 func getJSONTreeValForPath(tree interface{}, path []string) (interface{}, bool) {
-	t := tree
-	for i := 0; i < len(path); i++ {
-		tt, ok := t.(map[string]interface{})
-		if !ok {
-			return nil, false
-		}
-		nv, ok := mapValueNoPrefix(tt, path[i])
-		if !ok {
-			return nil, false
-		}
-		t = nv
+	if len(path) == 0 {
+		return tree, true
 	}
-	dbgPrint("path:%v matches tree %s, return subtree %s", path, valueStr(tree), valueStr(t))
 
-	return t, true
+	t, ok := tree.(map[string]interface{})
+	if !ok {
+		return nil, false
+	}
+
+	for k, v := range t {
+		if path[0] == stripModulePrefix(k) {
+			ret, ok := getJSONTreeValForPath(v, path[1:])
```

Comment:
Consider collapsing the if within this block to be `if ret, ok := getJSONTreeValForPath(...); ok { }`.

Reply:
Done.